src/components/Proposal/Create/PluginForms/CreateUnknownGenericPluginProposal.tsx

```diff
@@ -72,12 +89,23 @@ class CreateGenericPlugin extends React.Component<IProps, IStateProps> {
       () => Object.assign(this.currentFormValues, this.state),
       (formValues: IFormValues, firstTime: boolean) => {
         this.currentFormValues = formValues;
-        if (firstTime) { this.state = { tags: formValues.tags }; }
-        else { this.setState({ tags: formValues.tags }); }
+        if (firstTime) {
+          Object.assign(this.state, {
+            tags: formValues.tags, abiInputs: this.state.abiInputs, callData: this.state.callData,
+          });
+        }
+        else { this.setState({ tags: formValues.tags, abiInputs: [], callData: "" }); }
       },
       this.props.showNotification);
   }
 
+  async componentDidMount() {
+    const contractToCall = (this.props.pluginState as IGenericPluginState).pluginParams.contractToCall;
+    const ABI_DATA = await getABIByContract(contractToCall);
```

Comment:
what if `getABIByContract` return undefined ?

Reply:
I can update `getABIByContract` to return an empty array instead of `undefined` along with the current console message `Failed to retrieve ABI`

Reply:
so any how you will need to check the return data of `getABIByContract`

Reply:
In case we get an empty array, a `No ABI for this contract` message is displayed instead of the form.